ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -147,34 +146,6 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
         return new IngestionStatusResult(ingestionStatus);
     }
 
-    private String getFormat(IngestionProperties ingestionProperties) {
-        String format = ingestionProperties.getDataFormat();
-        if (format == null) {
-            return "csv";
-        }
-        return format;
-    }
-
-    private String getMappingReference(IngestionProperties ingestionProperties, String format) throws IngestionClientException {
-        IngestionMapping ingestionMapping = ingestionProperties.getIngestionMapping();
-        String mappingReference = ingestionMapping.getIngestionMappingReference();
-        IngestionMapping.IngestionMappingKind ingestionMappingKind = ingestionMapping.getIngestionMappingKind();
-        if (IngestionMapping.mappingRequiredFormats.contains(format)) {
-            String message = null;
-            if (!format.equalsIgnoreCase(ingestionMappingKind.name())) {
-                message = String.format("Wrong ingestion mapping for format %s, found %s mapping kind.", format, ingestionMappingKind.name());
-            }
-            if (StringUtils.isBlank(mappingReference)) {
-                message = String.format("Mapping reference must be specified for %s format.", format);
```

Comment:
This is not identical to the checks in `IngestionProperties.validate()`. There, `columnMappings` can be provided instead of a `mappingReference` and not fail this. The question is what is correct - both in all cases, and whether `StreamingClient` is a special case?

@ArielYehezkely ?

Reply:
I wrote on the other comment before I saw this. StreamingIngestClient can only have mappingReference, so this check is correct.

Reply:
Already addressed the fact that Streaming is a unique case, - if mapping is required, it must be a reference.